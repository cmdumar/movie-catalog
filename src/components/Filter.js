import PropTypes from 'prop-types';
import styles from './Filter.module.scss';

const Filter = ({ handleSelect, category }) => (
  <div className={styles.filter}>
    <h3>Choose Category:</h3>
    <select value={category} onChange={handleSelect}>
      <option value="popular">Popular</option>
      <option value="now_playing">Now Playing</option>
      <option value="top_rated">Top Rated</option>
      <option value="upcoming">Upcoming</option>
    </select>
  </div>
);

Filter.propTypes = {
  handleSelect: PropTypes.func.isRequired,
  category: PropTypes.string.isRequired,
};

export default Filter;
