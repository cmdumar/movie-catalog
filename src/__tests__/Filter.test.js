import renderer from 'react-test-renderer';
import Filter from '../components/Filter';

test('Render Filter Component', () => {
  const f = renderer.create(<Filter handleSelect={() => 'something'} category="popular" />).toJSON();
  expect(f).toMatchSnapshot();
});
