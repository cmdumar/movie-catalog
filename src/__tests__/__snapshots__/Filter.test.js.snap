// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Filter Component 1`] = `
<div
  className="filter"
>
  <h3>
    Choose Category:
  </h3>
  <select
    onChange={[Function]}
    value="popular"
  >
    <option
      value="popular"
    >
      Popular
    </option>
    <option
      value="now_playing"
    >
      Now Playing
    </option>
    <option
      value="top_rated"
    >
      Top Rated
    </option>
    <option
      value="upcoming"
    >
      Upcoming
    </option>
  </select>
</div>
`;
